src/librustdoc/clean/mod.rs

```diff
@@ -1370,6 +1370,9 @@ pub(crate) fn clean_middle_assoc_item<'tcx>(
             generics.params.extend(late_bound_regions);
 
             let mut decl = clean_fn_decl_from_did_and_sig(cx, Some(assoc_item.def_id), sig);
+            if tcx.asyncness(assoc_item.def_id).is_async() {
+                decl.output = decl.sugared_async_return_type();
```

Comment:
Since this check and modification are done more than once, shouldn't this be moved directly into the `clean_fn_decl_from_did_and_sig` function?

Reply:
Yeah, good point. I considered that at some point but I waved it aside since we only have an `Option<DefId>` in `clean_fn_decl_from_did_and_sig` instead of a `DefId` since the ID isn't available for function point types.

However, thinking about it again, that argument doesn't make sense, I just need to match on it ^^. Gonna change and push that later :+1: